iis-common/src/main/java/eu/dnetlib/iis/common/java/io/SequenceFileTextValueReader.java

```diff
@@ -0,0 +1,168 @@
+                                return Pattern.matches(whitelistRegexp, 
+                                                fileStatus.getPath().getName());
+                        } else {
+                                return true;
+                        }
+                }
+//              fallback
+                return false;
+        }
+
+        /**
+         * Returns next data package.
+         * @return next data package
+         * @throws IOException
+         */
+        protected Text getNext() {
+                try {
+                        if (sequenceReader==null) {
+                                return null;
+                        }
+                        Writable key = (Writable) ReflectionUtils.newInstance(
+                                        sequenceReader.getKeyClass(), fs.getConf());
+                        Writable value = (Writable) ReflectionUtils.newInstance(
+                                        sequenceReader.getValueClass(), fs.getConf());
+                        if (sequenceReader.next(key, value)) {
+                        return (Text) value;
+                      } else {
+                          sequenceReader.close();
+                          sequenceReader = getNextSequenceReader();
+                          if (sequenceReader!=null) {
+                                  return getNext();
+                          }
+                      }
+//                      fallback
+                        return null;
+                } catch (IOException e) {
+                        throw new RuntimeException(e);
+                }
+        }
+
+        /* (non-Javadoc)
+         * @see java.util.Iterator#hasNext()
+         */
+        @Override
+        public boolean hasNext() {
+//              check and provide next when already returned
+                if (toBeReturned==null) {
+                        toBeReturned = getNext();
+                } 
+                return toBeReturned!=null;
+        }
+
+        /* (non-Javadoc)
+         * @see java.util.Iterator#next()
+         */
+        @Override
+        public Text next() {
+                if (toBeReturned!=null) {
+//                      element fetched while executing hasNext()
+                        Text result = toBeReturned;
+                        toBeReturned = null;
+                        return result;
+                } else {
+                        return getNext();
+                }
+        }
+
+        /* (non-Javadoc)
+         * @see eu.dnetlib.iis.exp.iterator.ClosableIterator#close()
+         */
+        @Override
+        public void close() {
+                if (sequenceReader!=null) {
+                        try {
+                                sequenceReader.close();
+                        } catch (IOException e) {
+                                log.error("error occurred when closing sequence reader", e);
+                        }
```

Comment:
Hmm, I don't recall why is this logged. 

Iterator does not seem to be the best place to log things anyway, added `throws` declaration instead.